### ISSUE TITLE
fails on `{lat:0,lon:0}, {lat:0,lon:180}`

### ISSUE THREAD

**Issue by stumash** [gabmontes/fast-haversine#2]
hey, really cool library, thanks for making and publishing it!

I get `NaN` if I do `distance({lat:0, lon:0}, {lat:0, lon:180})`. I'm no expert at geographic coords, but I feel like it's not supposed to be `NaN`.

**Comment by gabmontes** [comment]
@stumash thanks!

As mentioned in my original blog post (added a link to the README), in order to make it fast I cut some corners. One of the main assumptions is the two points in the sphere are close to each other. In the case you mentioned, that assumption is not valid. Therefore the result is not valid either.

If you need more precise distance calculations, please try a different library. And benchmark yourself. It looks like `haversine-distance` is doing amazingly fast nowadays.

**Comment by stumash** [comment]
Unfortunately, I didn't find your code from the blog post, just by googling.

I think you should mention the limitations in the Readme, just so that
other people in my situation don't find out the hard way (and waste 45m
debugging their code).

I'm not trying to be difficult, but maybe saying something like "returns
NaN for points more than X many degrees apart", or just an example showing
that it fails for some input.

Basically I think the Readme should mention the failure cases, or at the
very least that they exist (since it's a feature, not a bug). I can submit
a PR if you want, to save you some time.

**Comment by gabmontes** [comment]
I will add a note. Thanks for the comment.
